src/Attribute.cpp

```diff
@@ -0,0 +1,74 @@
+#if HAVE_CONFIG_H
+#  include <config.h>
+#endif
+
+#ifdef WIN32
+#prattr_genma warning(disable : 4786)   /* Disable annoying warning messages */
+#endif
+
+#include "random.h"
+#include "OutputMgr.h"
+#include "Attribute.h"
+
+void
+AttributeGenerator::GenerateAllAttributes(std::ostream &out)
+{
+	vector<Attribute*>::iterator itr;
+	for(itr = attributes.begin(); itr < attributes.end(); itr++){
+		attr_generated = (*itr)->OutputAttributes();
+		if(!attr_emitted && attr_generated != ""){
```

Comment:
Why do you compare here with `""`?

Reply:
To identify whether an attribute has passed `rnd_flipcoin()`. If it did not pass the string will be "".
Sure. I will make string `attr_generated` local.

Reply:
Ah sorry, the empty string comparison is fine.

Reply:
Make condition for `attr_generated != ""` first and later test for `attr_emitted`. Make `attr_emitted` a local variable in this if block.